js_modules/dagster-ui/packages/ui-core/src/asset-graph/CollapsedGroupNode.tsx

```diff
@@ -0,0 +1,115 @@
+import {Box, Colors, FontFamily, Icon} from '@dagster-io/ui-components';
+import React from 'react';
+import styled from 'styled-components';
+
+import {withMiddleTruncation} from '../app/Util';
+
+import {AssetDescription, NameTooltipCSS} from './AssetNode';
+import {GroupLayout} from './layout';
+
+export const GroupNodeNameAndRepo = ({group, minimal}: {minimal: boolean; group: GroupLayout}) => {
+  const name = `${group.groupName} `;
+  const location = `${group.repositoryName}@${group.repositoryLocationName}`;
+
+  if (minimal) {
+    return (
+      <Box style={{flex: 1, fontFamily: FontFamily.monospace}}>
+        <div
+          data-tooltip={name}
+          data-tooltip-style={GroupNameTooltipStyle}
+          style={{fontSize: 30, fontWeight: 600, lineHeight: '30px'}}
+        >
+          {withMiddleTruncation(name, {maxLength: 14})}
```

Comment:
Any reason for using `withMiddleTruncation` as opposed to the `MiddleTruncate` component which will measure the text for you and pick the appropriate length?

Reply:
Hmm I sorta did this throughout the asset graph because the fonts are monospace and the boxes have a fixed width, so all the observing/measuring would yield the same result for every box. Also slightly worried about rendering ~hundreds of the MiddleTruncate things since they each have an effect + resize observer + single state update after mount.